### DISCUSSION TITLE
update loop

### DISCUSSION THREAD

**Discussion by adsick** [nushell/nushell#3788] [Ideas]
Hello, I thought it is possible to make something like:
```
loop {ps; clear}
```
or 
```
loop {ps; sleep 0.5sec; clear}
```
but unfortunately there is no ```loop```
and with ```for i in 1..10 {ps; sleep 0.5sec; clear}``` the behavior is far from sweet.

I guess there could be some command to ```replace```, or ```update``` output of command execution with new result.
also ```clear``` command could take flags/arguments to clear only one line (or n lines) or something.

**Comment by jokeyrhyme** [comment]
I got something similar working with `yes | each { ps; sleep 5sec; clear }`

**Reply by adsick** [reply]
```^^^ command yes not found```

**Reply by jokeyrhyme** [reply]
Weird, that's part of the coreutils package: https://www.gnu.org/software/coreutils/manual/html_node/yes-invocation.html

My example should work if you can find a replacement for `yes` that run forever and outputs something over and over

**Reply by adsick** [reply]
so that is literally the problem of this issue

**Comment by dscottboggs** [comment]
I wrote this after stumbling on this

```nu
def loop [block: block] {
    open /dev/zero | each $block
}
```
then I can just do
~~~sh
$ loop { echo test; sleep 1sec }
~~~
